mgmt/rest/server.go

```diff
@@ -135,6 +139,35 @@ func New(https bool, cpath, kpath string) (*Server, error) {
 	return s, nil
 }
 
+// SetAPIAuth sets API authentication to enabled or disabled
+func (s *Server) SetAPIAuth(auth bool) {
+	s.auth = auth
+}
+
+// SetAPIAuthHashedPwd sets the API authentication password from snapd
+func (s *Server) SetAPIAuthHashedPwd(pwd []byte) {
+	s.authpwd = pwd
+}
+
+// Auth Middleware for REST API
+func (s *Server) authMiddleware(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	if s.auth {
+		_, password, ok := r.BasicAuth()
+		defer r.Body.Close()
+		err := bcrypt.CompareHashAndPassword(s.authpwd, []byte(password))
+		// If we have valid password or going to tribe/agreements endpoint
+		// go to next. tribe/agreements endpoint used for populating
+		// snapctl help page when tribe mode is turned on.
+		if (ok && err == nil) || r.RequestURI == "/v1/tribe/agreements" {
```

Comment:
I don't think we want to give access to the `/v1/tribe/agreements` endpoint, just so `snapctl` can check on Tribe availability.  I would rather handle this on the `snapctl` side, either by removing the feature toggling all together, or printing something to the screen about being unable to authenticate to the daemon and then printing the normal help output.